buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/codenarc/IntegrationTestFixturesRule.kt

```diff
@@ -0,0 +1,61 @@
+package org.gradle.gradlebuild.buildquality.codenarc
+
+import org.codehaus.groovy.ast.ClassNode
+import org.codehaus.groovy.ast.MethodNode
+import org.codehaus.groovy.ast.expr.MethodCallExpression
+import org.codehaus.groovy.ast.expr.PropertyExpression
+import org.codehaus.groovy.ast.expr.VariableExpression
+import org.codenarc.rule.AbstractAstVisitor
+import org.codenarc.rule.AbstractAstVisitorRule
+import org.codenarc.util.AstUtil
+
+
+class IntegrationTestFixturesRule : AbstractAstVisitorRule() {
+    override fun getName(): String = "IntegrationTestFixtures"
+
+    override fun getPriority(): Int = 1
+
+    override fun setPriority(priority: Int) {
+        throw UnsupportedOperationException()
+    }
+
+    override fun setName(name: String?) {
+        throw UnsupportedOperationException()
+    }
+
+    override fun getDescription(): String = "Reports incorrect usages of integration test fixtures"
+
+    override fun getAstVisitorClass(): Class<*> = IntegrationTestFixtureVisitor::class.java
+}
+
+
+class IntegrationTestFixtureVisitor : AbstractAstVisitor() {
+
+    /**
+     * Determines if the class is an integration test. It's a bit weak, but we cannot
+     * rely on the class node analysis since it doesn't give the right superclass because
+     * resolution is not complete when this visitor is executed.
+     */
+    private
+    fun isIntegrationTest(current: ClassNode) = current.name.endsWith("IntegrationTest")
+        || current.name.endsWith("IntegrationSpec")
+
+    override fun shouldVisitMethod(node: MethodNode?): Boolean = isIntegrationTest(node!!.declaringClass)
+
+    override fun visitMethodCallExpression(call: MethodCallExpression?) {
```

Comment:
:x: Kotlin declares the parameter type as nullable by default but you should change it to non-null if it's not expected to receive null.